domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java

```diff
@@ -461,7 +507,19 @@ private void init() throws Exception {
       startEmbeddedPostgres();
     }
 
-    client = io.vertx.ext.asyncsql.PostgreSQLClient.createNonShared(vertx, postgreSQLClientConfig);
+    client = null;
+    startPool();
+
+    log.debug("Setup timer = " + activityTimerId + " for tenant " + tenantId);
+    activityTimerId = vertx.setPeriodic(postgreSQLClientConfig.getInteger(CONNECTION_RELEASE_DELAY), res -> {
+      log.debug("expire timer for tenant " + tenantId);
+      if (!pgPoolActivity && client != null) {
```

Comment:
With `startPool` being called at an arbitrary time by `getConnection` , is it possible that the client might be closed immediately after the `client != null` check of `startPool`?

Reply:
I'm going to remove all this. Connections that are handed out by the pool are also closed, upon close of the pool.